### PR TITLE
Add support for entry_points as string

### PR DESCRIPTION
**Pull request by pikeas**
Fixes issue #239.

### CODE REVIEW

File: pex/commands/bdist_pex.py [hunk 1]
```diff
@@ -67,6 +68,10 @@ def split_and_strip(entry_point):
           for script in self.distribution.entry_points.get('console_scripts', []))
     except ValueError:
       console_scripts = {}
+    except AttributeError:
```
Comment (jsirois) [inline pex/commands/bdist_pex.py]: This is a bit cryptic / indirect - where is `AttributeError` raised from - takes a bit of staring.  How about being direct and restructuring to test `self.distribution.entry_points` with:
```python
if isinstance(self.distribution.entry_points, Compatibility.string): # See: https://github.com/pantsbuild/pex/blob/master/pex/compatibility.py#L35
  # ConfigParser
elif isinstance(self.distribution.entry_points, dict):
  # dict access
else:
  raise TypeError(...
```

**Comment by jsirois** [comment]
Thanks for taking this up!

**Comment by pikeas** [comment]
Pushed an update. Having some trouble running Tox locally, hopefully everything comes up clean.

**Comment by jsirois** [comment]
This LGTM.  Assuming a green CI run I'll patch this in.

**Comment by jsirois** [comment]
@pikeas looks like an `isort` pass is needed to fix import order.

**Comment by pikeas** [comment]
Yep, I see the failure, but I'm not able to reproduce locally, and Travis doesn't provide the correct sort order. I've tried running the test's isort command directly, but that generates a ton of noise.

**Comment by jsirois** [comment]
In case you decide to contribute more, like so:
```
$ curl -sSL https://github.com/pantsbuild/pex/pull/240.diff | git apply -3
$ tox -e isort-run
GLOB sdist-make: /home/jsirois/dev/pantsbuild/pex/setup.py
isort-run create: /home/jsirois/dev/pantsbuild/pex/.tox/isort-run
isort-run installdeps: isort
isort-run inst: /home/jsirois/dev/pantsbuild/pex/.tox/dist/pex-1.1.4.zip
isort-run installed: isort==4.2.5,pex==1.1.4
isort-run runtests: PYTHONHASHSEED='3672361260'
isort-run runtests: commands[0] | isort -ns __init__.py -rc /home/jsirois/dev/pantsbuild/pex/pex /home/jsirois/dev/pantsbuild/pex/tests
Skipped 3 files
__________________________________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________________________________
  isort-run: commands succeeded
  congratulations :)
```

And that resulted in this diff:
```diff
$ git diff -U1
diff --git a/pex/commands/bdist_pex.py b/pex/commands/bdist_pex.py
index cc82b85..bc98650 100644
--- a/pex/commands/bdist_pex.py
+++ b/pex/commands/bdist_pex.py
@@ -1,5 +1,5 @@
 import os
-from configparser import ConfigParser
 from distutils import log
 
+from configparser import ConfigParser
 from setuptools import Command
diff --git a/tests/test_inherits_path_option.py b/tests/test_inherits_path_option.py
index d38687b..ee8edc7 100644
--- a/tests/test_inherits_path_option.py
+++ b/tests/test_inherits_path_option.py
@@ -4,3 +4,2 @@
 import os
-
 from contextlib import contextmanager
```

**Comment by jsirois** [comment]
Thanks @pikeas - this is patched into master as 1 squashed commit with the isort fix applied and your original description as the commit message here: https://github.com/pantsbuild/pex/commit/c7651f02140e607caaaf2d4e007f196dd15d02c3

**Comment by pikeas** [comment]
Thanks for sharing the Tox command, @jsirois - it wasn't clear that isort-run was an environment!

**Pull request closed without merging**